extensions/amp-ad/0.1/amp-ad-xorigin-iframe-handler.js

```diff
@@ -94,6 +95,19 @@ export class AmpAdXOriginIframeHandler {
         .then(info => {
           this.element_.creativeId = info.data.id;
         });
+    listenFor(this.iframe, 'get-html', (info, source, origin) => {
+      if (!this.iframe) {
+        return;
+      }
+
+      const {selector, attributes} = info;
+      const content = getHtml(this.baseInstance_.win, selector, attributes);
```

Comment:
Would you be ok with requiring an attribute on the ad like `data-html-access-allowed` on the `amp-ad` element.

The current implementation allows all ads access to the HTML of the page. This may include user identifiers in expanded links. I think making that available to an ad, would need page consent.